WebApi/WebApi/Controllers/PostController.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using WebApi.Models.DTO;
+using WebApi.Models.DTO.PostDTOs;
+using WebApi.Services.Services_Interfaces;
+
+
+namespace WebApi.Controllers
+{
+    [ApiController]
+    [Route("api/[controller]")]
+    public class PostController : Controller
+    {
+        private readonly IPostService _postService;
+        private readonly ILogger<PostController> _logger;
+        public PostController(ILogger<PostController> logger , IPostService postService)
+        {
+            _logger = logger;
+            _postService = postService;
+        }
+
+
+        [HttpGet]
+        public IQueryable<PostDTO> GetAll([FromHeader] int userID)
+        {
+            return _postService.GetAll();
+        }
+
+        //Proponuje przenieść to do kontrolera user. Zniknie konflikt posts/userID i posts/postID
+        [HttpGet("{userID}")]
```

Comment:
W dokumentacji jest rozróżnienie na /post i /posts ale wg mnie to słabe
ja bym to rozróznienie zrobił tak:
public IQueryable<PostDTO> GetUserPosts([FromQuery] int UserID) wtedy wywołanie by było:
.../api/posts?userId=1
oraz
public IQueryable<PostDTO> GetUserPosts([FromRoute] int PostID)
.../api/posts/1
Czyli mniej więcej coś takiego co ty napisałeś
Trzeba się dopytać reszty grup i postanowić